ckanext/data_qld/resource_freshness/helpers/helpers.py

```diff
@@ -68,10 +69,32 @@ def check_resource_data(current_resource, updated_resource, context):
     # The context['resource_data_updated'] value will be used in the validator 'validate_nature_of_change_data'
     context['resource_data_updated'] = {
         'id': updated_resource.get('id'),
-        'data_updated': data_updated
+        'data_updated': data_updated,
+        'url_type': current_resource.get('url_type')
     }
 
 
+def get_resource_file_url(resource_dict):
+    """
+    Return resource file/url.
+    """
+    from ckan.model import Package
+
+    url = resource_dict.get('url', '')
+    if len(url.rsplit('/')) == 1:
+        pkg_id = resource_dict.get('package_id', '')
+        res_id = resource_dict.get('id', '')
+
+        return core_helpers.url_for(controller='package',
```

Comment:
@awset FYI toolkit has core_helpers.url_for method. Try to always use the toolkit instead.